ClosureCompilerRecursive.java

```diff
@@ -0,0 +1,81 @@
+import java.io.IOException;
+import java.io.BufferedWriter;
+import java.io.FileWriter;
+import java.io.PrintWriter;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.nio.file.Path;
+import java.util.List;
+import java.util.ArrayList;
+
+import com.google.javascript.jscomp.CompilationLevel;
+import com.google.javascript.jscomp.Compiler;
+import com.google.javascript.jscomp.CompilerOptions;
+import com.google.javascript.jscomp.SourceFile;
+import com.google.javascript.jscomp.CommandLineRunner;
+import com.google.javascript.jscomp.CompilerOptions.LanguageMode;
+
+public class ClosureCompilerRecursive {
+   protected Integer errorCount = 0;
+
+   public String compile(String filename, CompilerOptions options) {
+      Compiler compiler = new Compiler();
+
+      List<SourceFile> list = null;
+
+      try {
+          list =
+         CommandLineRunner
+         .getBuiltinExterns(CompilerOptions.Environment.BROWSER);
+      } catch (IOException e) {
+          System.out.println(e.getMessage());
+          System.exit(1);
+      }
+
+      list.add(SourceFile.fromFile(filename));
+      compiler.compile(new ArrayList<SourceFile>(), list, options);
+      errorCount += compiler.getResult().errors.length;
+      return compiler.toSource();
+   }
+
+   public static void main(String[] args) throws IOException {
+      ClosureCompilerRecursive ccr = new ClosureCompilerRecursive();
+      ccr.compileDirectory(args[0]);
+      System.exit(Math.min(ccr.errorCount, 127));
+   }
+
+   protected void compileDirectory(String dir) throws IOException {
+      CompilerOptions options = getOptions();
+      for(Path file : getFilesFromDirectory(dir)) {
+         String code = compile(file.toString(), options);
+         PrintWriter writer = new PrintWriter(new BufferedWriter(new FileWriter(file.toString())));
+         writer.println(code);
+         writer.close();
+         System.gc();
+      }
+   }
+
+   protected Path[] getFilesFromDirectory(String dir) throws IOException {
+      return Files.find(
+         Paths.get(dir),
+         Integer.MAX_VALUE,
+           (path, fileAttr) -> fileAttr.isRegularFile())
```

Comment:
Oo anonymous function in Java

Reply:
Yeah, all the languages are converging, it's pretty fun.